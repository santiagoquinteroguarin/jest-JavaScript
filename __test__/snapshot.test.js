import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
    test('snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('Instantánea con algunas Excepciones en sus paths', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Santiago Quintero',
            profesion: 'JavaScript Development',
            createdAt: new Date
        };
    
        // A excepción de los campos id y createdAt, los cuales se espera que reciban cualquier Numero y Fecha respectivamente
        expect(user).toMatchSnapshot({
            id: expect.any(Number),
            createdAt: expect.any(Date)
        });
    });
});
