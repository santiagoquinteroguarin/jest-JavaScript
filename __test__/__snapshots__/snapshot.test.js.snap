// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
